ui/app/components/secret-engine/configure-azure.ts

```diff
@@ -0,0 +1,185 @@
+  @service declare readonly store: StoreService;
+  @service declare readonly version: VersionService;
+  @service declare readonly flashMessages: FlashMessageService;
+
+  @tracked accessType = 'azure';
+  @tracked errorMessage = '';
+  @tracked invalidFormAlert = '';
+  @tracked saveIssuerWarning = '';
+
+  disableAccessType = false;
+
+  constructor(owner: unknown, args: Args) {
+    super(owner, args);
+    // the following checks are only relevant to existing enterprise configurations
+    if (this.version.isCommunity && this.args.model.isNew) return;
+    const { isWifPluginConfigured, isAzureAccountConfigured } = this.args.model;
+    this.accessType = isWifPluginConfigured ? 'wif' : 'azure';
+    // if there are either WIF or azure attributes, disable user's ability to change accessType
+    this.disableAccessType = isWifPluginConfigured || isAzureAccountConfigured;
+  }
+
+  get modelAttrChanged() {
+    // "backend" dirties model state so explicity ignore it here
+    return Object.keys(this.args.model?.changedAttributes()).some((item) => item !== 'backend');
+  }
+
+  get issuerAttrChanged() {
+    return this.args.issuerConfig?.hasDirtyAttributes;
+  }
+
+  @action continueSubmitForm() {
+    this.saveIssuerWarning = '';
+    this.save.perform();
+  }
+
+  // check if the issuer has been changed to show issuer modal
+  // continue saving the configuration
+  submitForm = task(
+    waitFor(async (event: Event) => {
+      event?.preventDefault();
+      this.resetErrors();
+      const { issuerConfig } = this.args;
+
+      if (issuerConfig?.hasDirtyAttributes) {
+        // if the issuer has changed show modal with warning that the config will change
+        // if the modal is shown, the user has to click confirm to continue saving
+        this.saveIssuerWarning = `You are updating the global issuer config. This will overwrite Vault's current issuer ${
+          issuerConfig.queryIssuerError ? 'if it exists ' : ''
+        }and may affect other configurations using this value. Continue?`;
+        // exit task until user confirms
+        return;
+      }
+      await this.save.perform();
+    })
+  );
+
+  save = task(
+    waitFor(async () => {
+      const modelAttrChanged = this.modelAttrChanged;
+      const issuerAttrChanged = this.issuerAttrChanged;
+      // check if any of the model or issue attributes have changed
+      // if no changes, transition and notify user
+      if (!modelAttrChanged && !issuerAttrChanged) {
+        this.flashMessages.info('No changes detected.');
+        this.transition();
+        return;
+      }
+
+      const modelSaved = modelAttrChanged ? await this.saveModel() : false;
+      const issuerSaved = issuerAttrChanged ? await this.updateIssuer() : false;
+
+      if (modelSaved || (!modelAttrChanged && issuerSaved)) {
+        // transition if the model was saved successfully
+        // or transition if no attributes on the model have changed and the issuer was saved successfully
```

Comment:
I wonder if this would be clearer as "We only prevent a transition if the model is edited and fails saving." Since the comment explains when we _do_ transition which we can see from the conditional 💬